testsuite/tests/with/shared-switch/test.py

```diff
@@ -0,0 +1,82 @@
+"""
+Check that selective sharing works as expected both locally and globally, and
+also when global sharing is enabled through the 'dependencies.dir' config.
+"""
+
+import os
+import shutil
+
+from drivers.alr import init_local_crate, run_alr
+from drivers.asserts import match_deploy_dir
+
+startdir = os.getcwd()
+global_cache = "alr-config"
+custom_cache = "custom_deps"
+
+def reset():
+    # Reset to initial test state
+    os.chdir(startdir)
+    shutil.rmtree("xxx", ignore_errors=True) # May not exist
+    shutil.rmtree("yyy", ignore_errors=True) # May not exist
+
+
+def local_cache(crate : str = "xxx") -> str:
+    """
+    Return the partial path to the local cache for a crate
+    """
+    return os.path.join(crate, "alire", "cache", "dependencies")
+
+
+# Check usual, simplest way of requesting a shared dependency
+init_local_crate()
+run_alr("with", "--shared", "libhello")
+match_deploy_dir("libhello", global_cache)
+reset()
+
+# Check that explicit switch works the same
+init_local_crate()
+run_alr("with", "--shared=yes_local", "libhello")
+match_deploy_dir("libhello", global_cache)
+reset()
+
+# Check global sharing hasn't occurred unexpectedly
+init_local_crate()
+run_alr("with", "libhello")
+match_deploy_dir("libhello", local_cache())
+reset()
+
+# Check global caching now
+init_local_crate()
+run_alr("with", "--shared=yes_global", "libhello")
+os.chdir(startdir)
+init_local_crate("yyy")
+run_alr("with", "libhello")
+match_deploy_dir("libhello", global_cache)
+reset()
+
+# Check that global caching can be prevented locally
+os.chdir(startdir)
+init_local_crate("yyy")
+run_alr("with", "--shared=no_local", "libhello")
```

Comment:
If I understand correctly, `with --shared=no_local  libhello` means that for this crate/workspace `libhello` should be deployed locally and not use the global shared deployment.

If I am correct, I think the command line is confusing as `no_local` means the "local" deployment will be used.

Reply:
Ah, you're right. Indeed, the idea is that the 4 combos are [yes|no] x [locally|globally], but now I see how it's misleading.